### PR TITLE
Explanations of Model Binding Default Behavior

### PR DESCRIPTION
**Pull request by anonymousgituser**
[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/mvc/models/model-binding?branch=pr-en-us-4830#how-model-binding-works)

# Explanations of Model Binding Default Behavior

* Summary

  This PR contributes to the article on model binding to further explain the default behavior for 
  parameters when values are not found in the request.

* Details

  The additions further explain what happens when values are not found for a given parameter. This addition details the default behavior based on a parameter's type.

* Suggested Reviewers

  Current authors: @Rick-Anderson, @isaac2004, @ardalis, @v-anpasi, and @rhires.

### CODE REVIEW

File: aspnetcore/mvc/models/model-binding.md [hunk 1]
```diff
@@ -57,7 +57,17 @@ so that the form fields remain filled with the user's input for their convenienc
...
+
+* `T[]`: With the exception of arrays of type `byte[]`, binding sets parameters of type `T[]` to `Array.Empty<T>()`. Arrays of type `byte[]` are set to `null`.
+
+* Reference Types: Binding creates an instance of a class with the default constructor without setting properties. However, model binding sets `string` parameters to `null`.
+
+* Nullable Types: Nullable types are set to `null`. In the above example, model binding sets `id` to `null` since it is of type `int?`.
+
+* Value Types: Non-nullable value types of type `T` are set to `default(T)`. For example, model binding will set a parameter `int id` to 0. You should consider using model validation or nullable types rather than relying on default values.
+
+If binding fails, MVC does not throw an error. You can query for model state errors by checking the `ModelState.IsValid` property.
```
Comment (rynowak) [inline aspnetcore/mvc/models/model-binding.md]: Consider making this statement stronger. **Every** action that takes user input should check `ModelState.IsValid.`

Comment (rynowak) [reply aspnetcore/mvc/models/model-binding.md]: Your statement isn't wrong 😆 But this is something you really need to know about to use MVC correctly

**Comment by scottaddie** [comment]
@anonymousgituser Thank you for taking the time to clarify how model binding works!

**Comment by anonymousgituser** [comment]
@scottaddie Happy to help!

**Pull request merged**